api/src/services/hashnode.ts

```diff
@@ -1,19 +1,59 @@
 import { request } from 'graphql-request'
-import gql from 'graphql-tag'
+import type { Connection } from 'types/connections'
+import type {
+  Post,
+  Publication,
+  Query,
+  QueryrecentPostsArgs as QueryRecentPostsArgs,
+} from 'types/graphql'
 
-export const recentPosts = async () => {
-  const query = gql`
+import { logger } from 'src/lib/logger'
+
+type RecentPostsResponse = {
+  publication: {
+    isTeam: Publication['isTeam']
+    title: Publication['title']
+    posts: Connection<Post>
+  }
+}
+
+/**
+ * Fetches a Publication with latest posts from Hashnode.
+ *
+ * Note: This resolver is cached for a specified amount of time
+ * defined in the GraphQL Handler in ttlPerSchemaCoordinate.
+ **/
+export const recentPosts = async ({
+  first = 3,
+}: QueryRecentPostsArgs): Promise<Query['recentPosts']> => {
+  const RECENT_POSTS = `
     {
       publication(host: "selfteach.me") {
         isTeam
         title
-        posts(first: 10) {
+        posts(first: ${first}) {
           edges {
             node {
               id
+              slug
               title
+              subtitle
               brief
+              content {
+                markdown
```

Comment:
Probably don't need all three, just maybe the markdown to render somewhere and maybe the brief on the feed list.